src/main/java/org/opensrp/service/OrganizationService.java

```diff
@@ -146,12 +147,17 @@ public void assignLocationAndPlan(String identifier, String jurisdictionId, Stri
 	 * @param returnFutureAssignments flag to control if future assignments are returned
 	 * @return the assigned locations and plans
 	 */
-	public List<AssignedLocations> findAssignedLocationsAndPlans(String identifier,boolean returnFutureAssignments) {
-		validateIdentifier(identifier);
-		Organization organization = getOrganization(identifier);
+	public List<AssignedLocations> findAssignedLocationsAndPlans(AssignedLocationAndPlanSearchBean assignedLocationAndPlanSearchBean) {
+		Organization organization = null;
+		if(assignedLocationAndPlanSearchBean != null && assignedLocationAndPlanSearchBean.getOrganizationIdentifier() != null) {
+			validateIdentifier(assignedLocationAndPlanSearchBean.getOrganizationIdentifier());
+			organization = getOrganization(assignedLocationAndPlanSearchBean.getOrganizationIdentifier());
+		}
 		if (organization == null)
 			throw new IllegalArgumentException("Organization not found");
-		return organizationRepository.findAssignedLocations(organization.getId(),returnFutureAssignments);
+
+		assignedLocationAndPlanSearchBean.setOrganizationId(organization.getId());
+		return organizationRepository.findAssignedLocations(assignedLocationAndPlanSearchBean);
```

Comment:
To make it easy to understand this method, can we retain the original contract and just add the pagination params. You can package the params back when invoking the repository

Reply:
Noted.